packages/client/hmi-client/src/services/charts.ts

```diff
@@ -42,6 +48,28 @@ export interface ErrorChartOptions extends Omit<BaseChartOptions, 'height' | 'yA
 	variables: { field: string; label?: string }[];
 }
 
+export interface InterventionMarkerOptions {
+	hideLabels?: boolean;
+	labelXOffset?: number;
+	dateOptions?: DateOptions;
+}
+
+function transformDateFormatFn(
+	date: Date,
+	datum: string,
+	type: 'year' | 'yearmonth' | 'yearmonthdate' = 'yearmonthdate'
```

Comment:
Any particular reason why we have two sets of time-measurement? Possible to use a single set of controlled vocabulary?
- CalendarDateType.DATE, CalendarDateType.MONTH, CalendarDateType.YEAR
- yearmonthdate, yearmonth, year

Reply:
I wasn't able to find a good way to use the same enum across the different 3rd party components we use.

These `CalendarDateType` enum is being used for the Primevue Calendar view https://v3.primevue.org/calendar/#api.calendar.props.view while the 'yearmonthdate', 'yearmonth', 'year' are for the vega chart https://vega.github.io/vega-lite/docs/timeunit.html definitions.

Reply:
This is no longer needed since the underlying chart is now using all number as discussed in our discussion late yesterday